### PR TITLE
batches: Add Mount Model

### PR DESCRIPTION
**Pull request by Piszmog**
Part of #31790.

Simply adding the `Mount` model to the batches lib and the `batch_spec.schema.json`.

Follow up PRs will come after merge.

## Test plan

Nothing to test, just adding models/structs.

### CODE REVIEW

File: schema/batch_spec.schema.json [hunk 1]
```diff
@@ -211,6 +211,27 @@
               "${{ outputs.goModFileExists }}",
               "${{ eq previous_step.stdout \"success\" }}"
             ]
+          },
+          "mount": {
```
Comment (eseliger) [inline schema/batch_spec.schema.json:215]: In the SSBC editor, maybe we should dynamically strip out this property, so it's not in the auto suggestions.

**Review by eseliger — APPROVED** [review]
Let's mount 🐴

File: schema/batch_spec.schema.json (hunk 1 above)
Comment (Piszmog) [reply schema/batch_spec.schema.json:215]: Updated `MonacoBatchSpecEditor.tsx` to delete the `mount` field.

**Review by eseliger — APPROVED** [review]

File: schema/batch_spec.schema.json [hunk 2]
```diff
@@ -211,6 +211,27 @@
...
+              "required": ["path", "mountpoint"],
+              "properties": {
+                "path": {
+                  "type": "string",
+                  "description": "The path on the local machine to mount. The path must be in the same directory or a subdirectory of the Batch Change Specification.",
+                  "examples": ["local/path/to/file.text", "local/path/to/directory"]
+                },
+                "mountpoint": {
+                  "type": "string",
+                  "description": "The path on the container to mount the path on the local machine to.",
```
Comment (LawnGnome) [inline schema/batch_spec.schema.json:230]: Absolute nitpickery:

```suggestion
                  "description": "The path in the container to mount the path on the local machine to.",
```

**Review by courier-new — APPROVED** [review]

**Pull request merged**